test/e2e/framework/conformance/conformance.go

```diff
@@ -0,0 +1,845 @@
+		options := metav1.PatchOptions{FieldValidation: "Strict"}
+		switch patch.Type {
+		case types.ApplyYAMLPatchType, types.ApplyCBORPatchType:
+			options.FieldManager = "test-apply"
+			options.Force = ptr.To(true)
+		}
+		existingObject, err = client.Patch(ctx, desiredInitialObject.GetName(), patch.Type, patchData, options, "status")
+		framework.ExpectNoError(err, "patch %s %s status", patch.Type, resourceType)
+		verifyObject(fmt.Sprintf("patch %s status", patch.Type), desiredUpdatedObjectWithStatus, existingObject)
+	}
+
+	// Use the label as selector in List and DeleteCollection calls.
+	ginkgo.By(fmt.Sprintf("Listing %s collection with label selector %s", gvr, listOptions.LabelSelector))
+	items, err := client.List(ctx, listOptions)
+	framework.ExpectNoError(err, "list %s", resourceType)
+	gomega.Expect(items.Items).Should(gomega.HaveLen(1), "Should have listed exactly the test resource.")
+	verifyObject("list", desiredUpdatedObject, &items.Items[0])
+
+	if tc.IsNamespaced() {
+		ginkgo.By(fmt.Sprintf("Listing %s without namespace and with label selector %s", gvr, listOptions.LabelSelector))
+		items, err := resourceClient.List(ctx, listOptions)
+		framework.ExpectNoError(err, "list %s in all namespaces", resourceType)
+		gomega.Expect(items.Items).Should(gomega.HaveLen(1), "Should have listed exactly the test resource in all namespaces.")
+		verifyObject("list all namespaces", desiredUpdatedObject, &items.Items[0])
+	}
+
+	ginkgo.By(fmt.Sprintf("Deleting %s collection with label selector %s", gvr, listOptions.LabelSelector))
+	err = client.DeleteCollection(ctx, metav1.DeleteOptions{}, listOptions)
+	framework.ExpectNoError(err, "delete collection of %s", resourceType)
+	ensureNotFound(ctx, getResource)
+}
+
+// ensureNotFound ensures that the error is NotFound.
+func ensureNotFound(ctx context.Context, get func(context.Context) (*unstructured.Unstructured, error)) {
+	ginkgo.GinkgoHelper()
+	ginkgo.By("Checking for existence")
+	gomega.Eventually(ctx, func(ctx context.Context) error {
+		obj, err := framework.HandleRetry(get)(ctx)
+		switch {
+		case apierrors.IsNotFound(err):
+			return nil
+		case err != nil:
+			return fmt.Errorf("unexpected error after GET: %w", err)
+		default:
+			return fmt.Errorf("resource not removed yet:\n%s", marshalYAML(obj, 1))
+		}
+	}).WithTimeout(30 * time.Second /* From prior conformance tests, e.g. https://github.com/kubernetes/kubernetes/blame/be361a18dda0f2fab1f5e25f8067a9ed43fc3b89/test/e2e/storage/storageclass.go#L152 */).
+		Should(gomega.Succeed())
+}
+
+// marshalYAML produces multi-line YAML with each line indented by a certain
+// number of spaces (currently three) multiplied by the requested indention
+// level.
+//
+// Gomega could be configured to log unstructured.Unstructured as YAML,
+// but right now it isn't, so we have to do that ourselves instead of
+// using format.Object.
+func marshalYAML(obj any, indent int) string {
```

Comment:
nit: clarify in the method name this is only for logging, not for making yaml to then read / consume

```suggestion
func marshalYAMLForLogging(obj any, indent int) string {
```

Reply:
Done.